src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -662,6 +662,11 @@ private function handleOutOfStockProduct(OrderForViewing $orderForViewing)
      */
     public function addProductAction(int $orderId, Request $request): Response
     {
+        /** @var OrderForViewing $orderForViewing */
+        $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+
+        $previousProducts = $orderForViewing->getProducts()->getProducts();
```

Comment:
@jolelievre `getProducts()->getProducts()`? 😱

Reply:
Don't ask why, I already complained about it :p

Reply:
@PierreRambaud gosh :( can't we... do an alias for this...? good luck with getting back to this part of code in few months :/

Reply:
Ok, nvm, I see a structure now 😿

Reply:
I agree with you the structure doesn't make much sense 🤔
The extra layer `OrderProductsForViewing` seems quite useless to me.. Sadly I'm afraid it's a bit late to change it Although a small refacto PR could probably be proposed and be validate with Behat tests only

But I'm afrais I already know the answer if we propose this now 😅

Reply:
I guess the idea is that OrderForViewing is split into several pieces: `OrderProductsForViewing` `OrderShippingForViewing` and so on While it makes sens for classes that hold both a collection of sub entities and additional info (total count, recycled, ...) For products it does seem redundant 🤷‍♂️
But I see the logic now, at the very least the first getter should have been name differently 😛